src/main/java/io/stargate/sgv2/jsonapi/service/schema/collections/CollectionIdType.java

```diff
@@ -0,0 +1,36 @@
+package io.stargate.sgv2.jsonapi.service.schema.collections;
+
+import io.stargate.sgv2.jsonapi.exception.ErrorCodeV1;
+
+/** Collection Id Type enum, UNDEFINED represents unwrapped id */
+public enum CollectionIdType {
+  OBJECT_ID,
+  UUID,
+  UUID_V6,
+  UUID_V7,
+  UNDEFINED;
+
+  // TODO: store the name of the enum in the enum itself
```

Comment:
Since this PR is for refactoring, can this be done in this PR itself?

Reply:
when doing these re-orgs I try to not change the code much, created https://github.com/stargate/data-api/issues/1457 to track